src/common/common_init.cc

```diff
@@ -132,4 +133,23 @@ void common_init_finish(CephContext *cct)
       (cct->get_set_uid() || cct->get_set_gid())) {
     cct->get_admin_socket()->chown(cct->get_set_uid(), cct->get_set_gid());
   }
+
+  md_config_t *conf = cct->_conf;
+
+  if (!conf->admin_socket.empty() && !conf->admin_socket_mode.empty()) {
+    int ret = 0;
+    std::string err;
+
+    ret = strict_strtol(conf->admin_socket_mode.c_str(), 8, &err);
+    if (err.empty()) {
+      if (!(ret & (~07777))) {
```

Comment:
could use `ALLPERMS` or better off using `ACCESSPERMS`. for example, what is the use case of applying `setuid`?